ouroboros-consensus/src/Ouroboros/Storage/VolatileDB/Impl.hs

```diff
@@ -428,25 +441,16 @@ nextFile :: forall h m blockId. IOLike m
          -> m (InternalState blockId h)
 nextFile HasFS{..} _err VolatileDBEnv{..} st = do
     hClose $ _currentWriteHandle st
-    case _nextWriteFiles st of
-        [] -> do
-            let file = filePath $ _nextNewFileId st
-            hndl <- hOpen file (AppendMode MustBeNew)
-            return $ st {
-                  _currentWriteHandle = hndl
-                , _currentWritePath   = file
-                , _currentWriteOffset = 0
-                , _nextNewFileId      = (_nextNewFileId st) + 1
-                , _currentMap         = Map.insert file FileInfo.empty (_currentMap st)
-            }
-        (file, size) : rest -> do
```

Comment:
This was basically dead code, so the _nextWriteFiles field is totally erased. In the past, we supported a number of policies related to files which where not full: We could for example first try to fill these files and then proceed to creating new ones. The `_nextWriteFiles` field kept track of this. However we later deleted these policies and kept the default one (append to an existing non-empty file only if it is the latest in the db).